runtime/lua/vim/lsp/handlers.lua

```diff
@@ -117,15 +117,31 @@ M['window/showMessageRequest'] = function(_, result)
 end
 
 --see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#client_registerCapability
-M['client/registerCapability'] = function(_, _, ctx)
+M['client/registerCapability'] = function(_, result, ctx)
   local client_id = ctx.client_id
-  local warning_tpl = 'The language server %s triggers a registerCapability '
-    .. 'handler despite dynamicRegistration set to false. '
-    .. 'Report upstream, this warning is harmless'
   local client = vim.lsp.get_client_by_id(client_id)
-  local client_name = client and client.name or string.format('id=%d', client_id)
-  local warning = string.format(warning_tpl, client_name)
-  log.warn(warning)
+  for _, reg in ipairs(result.registrations) do
+    if reg.method == "workspace/didChangeWatchedFiles" then
```

Comment:
I have been doing most of my manual testing with `gopls`, which only supports this feature when registering dynamically.

Reply:
Afaik it's not possible use `didChangeWatchedFiles` _without_ dynamic registration. So this seems like the way to go.